lesson_9/lesson_9_authentication.py

```diff
@@ -0,0 +1,271 @@
+from dataclasses import dataclass
+import json
+
+
+@dataclass
+class DataStorage:
+    path_to_storage: str = 'logins.json'
+
+
+class User:
+    def __init__(self, name, password, age, key):
+        self.name = name
+        self.password = password
+        self.age = age
+        self.key = key
+
+    def save(self, _delete=None):
+        if _delete is not None:
+            logins = delete_changed_user(_delete.name)
+        else:
+            logins = get_all_logins()
+        logins.append({
+            "name": self.name,
+            "password": self.password,
+            "age": self.age,
+            "key": self.key
+        })
+        with open(DataStorage.path_to_storage, 'w') as file:
+            json.dump(logins, file, indent=' ')
+
+    @staticmethod
+    def change_name():
+        check_logins = turn_logins_into_class_user()
```

Comment:
`change_name`, `change_age` and `change_password` starting the same. I can define another function that will check existing user and dont duplicate the code.